public_html/dataView/allfiles.php

```diff
@@ -53,13 +55,11 @@
 		$files = array();
 
 		foreach ($object['data'][$field_name]['files']['archive'] as $file) {
-
-			$files[] = sprintf("%s",$file['name']);
-
+			$files[] = $file['name'];
 		}
 
 		$files           = implode(" ",$files);
-		$destinationFile = sys_get_temp_dir()."/".time().".".$type;
+		$destinationFile = sprintf("%s/%s.%s",mfcs::config('mfcsDownloadAllDir'),time(),$type);
 
 		if ($type == "zip") {
```

Comment:
That might make it more performant (likely would), but that should be addressed in a separate pull request. 

Doing that would not address the problem that the temporary directory isn't configurable, which this does.